### ISSUE TITLE
"Need to re-install WordPress?" should include dismissed updates

### ISSUE THREAD

**Issue by ocean90** [obenland/shiny-updates#242]
**Current Trunk:**

"Update" is hidden:
<img width="1356" alt="f" src="https://cloud.githubusercontent.com/assets/617637/15805105/af47e9ce-2b1f-11e6-8a32-d9a078afd007.png">

"Update" can be restored:
<img width="1413" alt="g" src="https://cloud.githubusercontent.com/assets/617637/15805107/b044d666-2b1f-11e6-8b4b-ee42c3f5de28.png">

**With Shiny Updates:**

Two updates:
<img width="1133" alt="a" src="https://cloud.githubusercontent.com/assets/617637/15805092/6917bde4-2b1f-11e6-8821-345a73d6e451.png">

Dismiss the localized update:
<img width="1142" alt="b" src="https://cloud.githubusercontent.com/assets/617637/15805094/69fa800c-2b1f-11e6-82e6-0e35c352eae5.png">

Update:
<img width="1131" alt="c" src="https://cloud.githubusercontent.com/assets/617637/15805095/6b2b5b54-2b1f-11e6-80ac-4111989bf7be.png">

No way to re-install:
<img width="1128" alt="d" src="https://cloud.githubusercontent.com/assets/617637/15805097/6c82f35e-2b1f-11e6-9e72-9011ad9a883e.png">

**Fix**

https://github.com/obenland/shiny-updates/blob/0207b88b9ab2c0f70d04fb18a154da807cbf8344/src/functions.php#L738 should be changed to `$core_updates = (array) get_core_updates( array( 'dismissed' => true ) );`:

<img width="1507" alt="e-fixed" src="https://cloud.githubusercontent.com/assets/617637/15805126/5e27aea2-2b20-11e6-9115-c9df56f39287.png">

An enhancement (which would make more sense) would be to provide the en_US re-install option if the localized update was dismissed. Or to show both options?

**Comment by swissspidy** [comment]
Thanks for the PR with your suggested fix 😛

> An enhancement (which would make more sense) would be to provide the en_US re-install option if the localized update was dismissed. Or to show both options?

Former would break existing behaviour, so I'd rather show both options, as we do in the updates table.

Other feedback welcome though. Pinging @mapk and @obenland for that.

**Comment by ocean90** [comment]
> Former would break existing behaviour,

Can you elaborate that?

**Comment by swissspidy** [comment]
Let me rephrase that. It would _change_ existing behaviour, as without Shiny Updates you can only re-install the localized update, not `en_US`. That's why I'd rather show both options.

By the way, I think we're currently missing the `This localized version contains both the translation and various other localization fixes.` and `There is a chance this update will break your translation. You may prefer to wait for the localized version to be released.` strings in Shiny Updates.

**Comment by ocean90** [comment]
> By the way, I think we're currently missing the…

The strings are kinda obsolete with the introduction of language packs...

**Comment by ocean90** [comment]
> > By the way, I think we're currently missing the…
>
> The strings are kinda obsolete with the introduction of language packs...

Looking at this again and found https://github.com/WordPress/WordPress/blob/c3055cc190a4b09131e12a88c7f904f9f1079066/wp-admin/includes/update-core.php#L991-L1012.

Translations are copied over if the update package includes the `wp-content/languages` directory. I'll talk with Dion about that. Maybe we should start to ship a localized `no-content` package.

**Comment by swissspidy** [comment]
`get_core_updates()` doesn't even return `en_US` and `de_DE` updates when running 4.5.2-de_DE.

Thus, 1cf67e50af92212f8de9dbb7f0ac0f0f936685c5 is probably enough for now, though I'd like to change the code for the re-install card to use a `foreach()` over the updates array instead of using `array_pop()`.

**Comment by swissspidy** [comment]
When running trunk in `de_DE`, it suggests re-installing `4.5.2-en_US`. Does that even make sense?
